### PR TITLE
fix(wsgi): ensure spans are finished when an unhandled BaseException is raised

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

- Ensure wsgi spans are closed if an Exception is raised in the application
- Add a comment to note when `wsgi.response` span is activated and closed. 

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: tests/contrib/wsgi/test_wsgi.py [hunk 1]
```diff
@@ -257,6 +259,15 @@ def test_500():
         app.get("/error")
 
 
+@snapshot(ignores=["meta.error.stack"], variants={"py2": PY2, "py3": PY3})
+def test_base_exception_in_wsgi_app():
```
Comment (Kyle-Verhoog) [inline tests/contrib/wsgi/test_wsgi.py:263]: :+1: great!

**Review by Kyle-Verhoog — DISMISSED** [review]

File: releasenotes/notes/fix-wsgi-exception-handling-fe30f2652b30933d.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+--- 
+fixes:
+  - |
+    wsgi: This fix resolves an issue where spans are not submitted when SystemExit is raised in a traced wsgi app.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/fix-wsgi-exception-handling-fe30f2652b30933d.yaml:4]:
```suggestion
    wsgi: This fix resolves an issue where ``BaseException``s raised in a WSGI application caused spans to not be submitted.
```

**Review by Kyle-Verhoog — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

File: releasenotes/notes/fix-wsgi-exception-handling-fe30f2652b30933d.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+--- 
+fixes:
+  - |
+    wsgi: This fix resolves an issue where ``BaseException``s raised in a WSGI application caused spans to not be submitted.
```
Comment (mabdinur) [inline releasenotes/notes/fix-wsgi-exception-handling-fe30f2652b30933d.yaml:4]: Failing test https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/24560/workflows/55b4f7da-78d3-4e16-8b81-76ca5edeed61/jobs/1665963

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by mabdinur** [comment]
@Mergifyio backport 1.6

**Pull request merged**